src/Parsers/IAST.h

```diff
@@ -248,6 +248,7 @@ class IAST : public std::enable_shared_from_this<IAST>, public TypePromotion<IAS
         bool surround_each_list_element_with_parens = false;
         bool ignore_printed_asts_with_alias = false; /// Ignore FormatState::printed_asts_with_alias
         bool allow_operators = true; /// Format some functions, such as "plus", "in", etc. as operators.
+        bool allow_moving_operators_before_parens = true;
```

Comment:
Yeah, sadly it is not possible to fix this formatting w/o introducing extra property (which I wanted to avoid in the first place)